### ISSUE TITLE
[Bug Report] ttnn.pad incorrect output shape for TILE Layout

### ISSUE THREAD

**Issue by mmanzoorTT** [tenstorrent/tt-metal#19513]
**Describe the bug**
ttnn.pad generates incorrect output tensor shape for TILE layout. 

**To Reproduce**
Python script to reproduce the bug
```
def test_pad():
  device_id = 0
  device = ttnn.open_device(device_id=device_id)
  ttnn.SetDefaultDevice(device)
  try:
    input = torch.ones(2, 8, dtype=torch.bfloat16)
    input_tensor = ttnn.from_torch(input, dtype=ttnn.bfloat16, device=device, layout=ttnn.TILE_LAYOUT)
    output = ttnn.pad(input1_tensor, [(0, 32), (0, 32)], value=5.0)
    out = ttnn.to_torch(output)
    print("Output: ", output)
    print("Torch.Output: ", out)
    print("Torch.output.shape: ", out.shape)

  finally:
      ttnn.close_device(device)
```

**Expected behavior**
Expected output tensor shape: `34x40`
Generated output tensor shape: `64x64`

**Screenshots**

**Please complete the following environment information:**
- OS: Ubuntu 22.04.4 LTS
- tt-metal commit `c11b5f3356`

**Additional context**

**Comment by mmanzoorTT** [comment]
@sdjordjevicTT

**Comment by mmanzoorTT** [comment]
tt-metal converts the input shape to padded shape and then add padding on top of it.

**Comment by amorrisonTT** [comment]
Hey @mmanzoorTT `ttnn::pad` support for tiled tensors is super awkward right now but it is, it can only pad in increments of tile size. This is bad but it isn't a bug per say
